src/java.base/share/classes/java/util/HexFormat.java

```diff
@@ -0,0 +1,1024 @@
+ * {@link java.io.Writer}, and {@link java.io.PrintStream}, all of which are {@link Appendable}.
+ * Each byte value is formatted as the prefix, two hexadecimal characters from the
+ * uppercase or lowercase digits, and the suffix.
+ * A delimiter appears after each formatted value, except the last.
+ * For conversions producing uppercase hexadecimal strings use {@link #withUpperCase()}.
+ *
+ * <p>
+ * For formatted hexadecimal string to byte array conversions the
+ * {@code parseHex} methods include {@link #parseHex(CharSequence) parseHex(string)} and
+ * {@link #parseHex(char[], int, int) parseHex(char[], offset, length)}.
+ * Each byte value is parsed as the prefix, two hexadecimal characters from the
+ * uppercase or lowercase digits, and the suffix.
+ * The delimiter is required after each formatted value, except the last.
+ *
+ * @apiNote
+ * For example, an individual byte is converted to a string of hexadecimal digits using
+ * {@link HexFormat#toHexDigits(int) toHexDigits(int)} and converted from a string to a
+ * primitive value using {@link HexFormat#fromHexDigits(CharSequence) fromHexDigits(string)}.
+ * <pre>{@code
+ *     HexFormat hex = HexFormat.of();
+ *     byte b = 127;
+ *     String byteStr = hex.toHexDigits(b);
+ *
+ *     int byteVal = hex.fromHexDigits(byteStr);
+ *     assert(byteStr.equals("7f"));
+ *     assert(b == byteVal);
+ *
+ *     // The hexadecimal digits are: "7f"
+ * }</pre>
+ * <p>
+ * For a comma ({@code ", "}) separated format with a prefix ({@code "#"})
+ * using lowercase hex digits the {@code HexFormat} is:
+ * <pre>{@code
+ *     HexFormat commaFormat = HexFormat.ofDelimiter(", ").withPrefix("#");
+ *     byte[] bytes = {0, 1, 2, 3, 124, 125, 126, 127};
+ *     String str = commaFormat.formatHex(bytes);
+ *
+ *     byte[] parsed = commaFormat.parseHex(str);
+ *     assert(Arrays.equals(bytes, parsed));
+ *
+ *     // The formatted string is: "#00, #01, #02, #03, #7c, #7d, #7e, #7f"
+ * }</pre>
+ * <p>
+ * For a fingerprint of byte values that uses the delimiter colon ({@code ":"})
+ * and uppercase letters the {@code HexFormat} is:
+ * <pre>{@code
+ *     HexFormat formatFingerprint = HexFormat.ofDelimiter(":").withUpperCase();
+ *     byte[] bytes = {0, 1, 2, 3, 124, 125, 126, 127};
+ *     String str = formatFingerprint.formatHex(bytes);
+ *     byte[] parsed = formatFingerprint.parseHex(str);
+ *     assert(Arrays.equals(bytes, parsed));
+ *
+ *     // The formatted string is: "00:01:02:03:7C:7D:7E:7F"
+ * }</pre>
+ *
+ * <p>
+ * This is a <a href="{@docRoot}/java.base/java/lang/doc-files/ValueBased.html">value-based</a>
+ * class; use of identity-sensitive operations (including reference equality
+ * ({@code ==}), identity hash code, or synchronization) on instances of
+ * {@code HexFormat} may have unpredictable results and should be avoided.
+ * The {@code equals} method should be used for comparisons.
+ *
+ * @implSpec
+ * This class is immutable and thread-safe.
```

Comment:
I'm curious why this is an implSpec, and not just a normative statement? HexFormat is a final class after all, there is just one implementation.

Reply:
Good point, will remove @ImplSpec.